inference-engine/src/transformations/src/transformations/common_optimizations/dropout_with_random_uniform_replacer.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/dropout_with_random_uniform_replacer.hpp"
+
+#include <memory>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::DropoutWithRandomUniformReplacer, "DropoutWithRandomUniformReplacer", 0);
+
+ngraph::pass::DropoutWithRandomUniformReplacer::DropoutWithRandomUniformReplacer() {
+    MATCHER_SCOPE(DropoutWithRandomUniformReplacer);
+    const auto shape_of_pattern = ngraph::pattern::wrap_type<opset8::ShapeOf>();
+    const auto ru_min_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    const auto ru_max_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    const auto random_uniform_pattern = ngraph::pattern::wrap_type<opset8::RandomUniform>(
+        {shape_of_pattern, ru_min_const_pattern, ru_max_const_pattern},
+        pattern::consumers_count(1));
+    const auto add_const_pattern = ngraph::pattern::wrap_type<opset8::Constant>();
+    const auto add_pattern = ngraph::pattern::wrap_type<opset8::Add>({random_uniform_pattern, add_const_pattern});
```

Comment:
what values do pattern's constants have? I am not sure that this transformation is valid for any constant values.
That is one more reason about why it is useful to have graphs in the description.

Reply:
For example, is the transformation valid for a case with min=10 and max=20 ?

Reply:
This transformation is not valid for min != 0 or max != 1. Added checks that min = 0, max = 1 and Add const has zero fractional part.